From 3f4fdf95dab54de16cc12c3651784028620bb33c Mon Sep 17 00:00:00 2001
From: Soumya Roy <souroy@souroy-mlt.client.nvidia.com>
Date: Mon, 31 Mar 2025 13:06:37 -0700
Subject: [PATCH] bgpd: Show bgp <afi> <safi> shouldn't display peers in groups
 The command `show bgp <afi> <safi>` has this output:

r1# show bgp ipv4 uni 10.0.0.0
BGP routing table entry for 10.0.0.0/32, version 1
Paths: (1 available, best #1, table default)
  Advertised to non peer-group peers:
  r1-eth0 r1-eth1 r1-eth2 r1-eth3
  ....

It specifically states `Advertised to non peer-group peers:` yet
the code is not filtering those out.

Signed-off-by: Donald Sharp <sharpd@nvidia.com>
---
 bgpd/bgp_route.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/bgpd/bgp_route.c b/bgpd/bgp_route.c
index 9e33bb32b..9902cd022 100644
--- a/bgpd/bgp_route.c
+++ b/bgpd/bgp_route.c
@@ -12013,6 +12013,9 @@ void route_vty_out_detail_header(struct vty *vty, struct bgp *bgp,
 	 * though then we must display Advertised to on a path-by-path basis. */
 	if (!bgp_addpath_is_addpath_used(&bgp->tx_addpath, afi, safi)) {
 		for (ALL_LIST_ELEMENTS(bgp->peer, node, nnode, peer)) {
+			if (peer->group)
+				continue;
+
 			if (bgp_adj_out_lookup(peer, dest, 0)) {
 				if (json && !json_adv_to)
 					json_adv_to = json_object_new_object();
-- 
2.43.2

